test-lib.sh

```diff
@@ -518,6 +533,29 @@ EOF
     fi
     # Filter out env var definitions from $s2i_args and create Dockerfile ENV commands out of them
     echo "$s2i_args" | grep -o -e '\(-e\|--env\)[[:space:]=]\S*=\S*' | sed -e 's/-e /ENV /' -e 's/--env[ =]/ENV /' >>"$df_name"
+    set +e
+    grep "NPM_MIRROR" "$df_name"
+    if [ $? -eq 0 ]; then
+        # NPM_MIRROR is in format
+        # http{s}://repository.engineering.redhat.com/nexus/repository/registry.npmjs.org
+        npm_mirror=$(cat $df_name | grep "NPM_MIRROR" | cut -d'=' -f2 | cut -d'/' -f3)
+        ping -c 5 $npm_mirror
+        if [ $? -ne 0 ]; then
+            echo "Internal '$npm_mirror' proxy server is not reachable."
+            echo "Do not copy certification file and use default 'registry.npmjs.org' registry."
+            # Delete NPM_MIRROR from Dockerfile
+            sed -ie "s/ENV NPM_MIRROR.*//" "$df_name"
+        else
+            # Check if CA file exists and update CA authorities in image.
+            ca_full_path="/etc/pki/ca-trust/source/anchors/RH-IT-Root-CA.crt"
+            if [ -e "$ca_full_path" ]; then
+                echo "RUN cd /etc/pki/ca-trust/source/anchors && update-ca-trust extract" >>"$df_name"
+            else
+              sed -ie "s/ENV NPM_MIRROR.*//" "$df_name"
+            fi
+        fi
+    fi
+    set -e
```

Comment:
This makes more sense as part of `ct_mount_ca_file` now.

Reply:
Or maybe a different function that could call `ct_mount_ca_file`, but it definitely should not be part of `ct_s2i_build_as_df`